### DISCUSSION TITLE
getting Top view after reopening the defaultfrontstage

### DISCUSSION THREAD

**Discussion by Rahul-0108** [iTwin/itwinjs-core#2985] [Q&A] [answered]
Hi , 
after opening custom frontstage and coming back to the main frontstage , I am not getting the the same view which I had before , insted I am getting top view .  I have attached a sandbox for this
https://www.itwinjs.org/sandboxes/MyPOC/views

why is this happening and any solution for it ??

**Comment by aruniverse** [comment, marked as answer]
The `viewConfiguration` callback in your sample is registered to only run once, [see](https://github.com/iTwin/viewer/blob/master/packages/modules/viewer-react/src/services/iModel/ViewCreator3d.ts#L47)

Here's a [sandbox](https://www.itwinjs.org/sandboxes/ArunGeorge/Josy-the-violet-clownfish-917) that should run the callback when you switch back from a different frontstage.

**Reply by Rahul-0108** [reply]
Thanks !!
